02-cpu-configure/ioc-restart.c

```diff
@@ -0,0 +1,87 @@
+/* BPM IOCs Utils
+ * Copyright (C) 2024 CNPEM
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+/*
+ * Slot whitelists to sanitize user input
+ */
+const char* const afc_whitelist[] = {
+    "1", "2", "2-1", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12"
+};
+
+const char* const rffe_whitelist[] = {
+    "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "23"
+};
+
+/*
+ * Check if the user provided slot string is valid
+ */
+int sanitize_slot(const char* slot, const char* const whitelist[], size_t whitelist_size) {
+    for (size_t i = 0; i < whitelist_size; i++) {
+        if (strcmp(slot, whitelist[i]) == 0) {
+            return 0;
+        }
+    }
+    return -1;
+}
+
+int main(int argc, char** argv) {
+    if (argc != 3) {
+        fprintf(stderr, "Usage: %s <afc|rffe> <slot_number>\n", argv[0]);
+        return 1;
+    }
+
+    const char* type = argv[1];
+    const char* slot = argv[2];
+
+    const char* const* whitelist = NULL;
+    size_t whitelist_size = 0;
+    char command[256];
+
+    /*
+     * Checks the IOC type and sets the whitelist
+     */
+    if (strcmp(type, "afc") == 0) {
+        whitelist = afc_whitelist;
+        whitelist_size = sizeof(afc_whitelist) / sizeof(afc_whitelist[0]);
```

Comment:
Is this correct? First value has a single byte but others have two.

Using integers and `atoi` wouldn't be easier to process?

Reply:
This is correct because doesn't matter the number of bytes, since `afc_whitelist` is an array of **pointers** to strings.
Using atoi or integers doesn't work in the case that the selected slot is '2-1', so it's easier to treat all values as strings.

Reply:
> afc_whitelist is an array of pointers to strings

That's fair.

> Using atoi or integers doesn't work in the case that the selected slot is '2-1'

Yeah, that would need to be treated differently, indeed. Well, that's fine to keep it the way it is.